### DISCUSSION TITLE
Security Concerns

### DISCUSSION THREAD

**Discussion by pjebs** [livewire/livewire#5248] [1. Help]
I was wondering exactly what mechanisms livewire was using to make sure its ajax requests couldn't be tampered with or modified by something other than the front-end application?

I'm just a bit hesitant to use the library without knowing.

**Comment by ahosker** [comment]
Have you read the [Livewire Security](https://laravel-livewire.com/docs/2.x/security) documentation?

> The fundamental security underpinning Livewire is a "checksum" that travels along with request/responses and is used to validate that the state from the server hasn't been tampered with in the browser.

**Reply by pjebs** [reply]
I read and understood the checksum bit.

I didn't fully understand the [Persistent Middleware](https://laravel-livewire.com/docs/2.x/security#persistent-middleware) section.
Perhaps the documentation can explain that in even greater detail for stupid people.

**Reply by ariaieboy** [reply]
when you visit a page exaple using `/admin` route you are gonna run some middleware for authentication and etc. so you can manage who can access your data. 
so when you load a page with livewire component if you want send and receive update for a specific component you don't need refresh your site cause livewire gonna handle it for you.
and for having this feature livewire using a route to send event's to the server and receive the updated data for that component. what persistence middleware do is applying the middleware that are being used in the base initial request so no one can copy a state of a component and trigger an event using that hidden route that handles the component update.

**Reply by ahosker** [reply]
Indeed - I think that is about Man In The Middle exploits.

**Comment by ariaieboy** [comment]
Regarding security, the livewire is secure enough. but it does not mean that any code that you wrote going to be safe. 
 for example, in mounted models, you must hide sensitive data in your model other than that all your model data gonna be publicly available on the front-end. or you can use Data Transfer Objects.

for some best practices about security and performance stuff please check out this repo https://github.com/michael-rubel/livewire-best-practices that's built by livewire communityy.

**Reply by levu42** [reply]
This link is awesome! It should be linked somewhere from the documentation or so, I had no idea about it - most of the things on the page I figured out myself over several weeks of developing a bigger application with Livewire, but some things on it were still new to me.

**Reply by ariaieboy** [reply]
yes, I learned some tips from this repo.
I think the best place to mention this Repo would be the awesome list unfortunately the `awesome-livewire` repo that has been mentioned in the livewire main Readme didn't get updated since 2020. I've reached out to the repo owner to give me access to maintain that repo and add new links to it. 
if he doesn't respond to me I may propose a new repo in the main Livewire organization

**Reply by alisalehi1380** [reply]
Tnks bro for sharing the repo. Very useful! 🔥
